src/devices/device.ts

```diff
@@ -82,16 +97,102 @@ export class Device extends EventEmitter {
         this.hub.send(data, characteristic, callback);
     }
 
-    public subscribe (mode: number) {
+    protected sendWithMode (modeName: string, data: Buffer, characteristic: string = Consts.BLECharacteristic.LPF2_ALL, callback?: () => void) {
+        const previousMode = this.mode;
+        this.subscribe(modeName);
+        this.send(data, characteristic, () => {
+            if (previousMode) {
+                this.subscribe(previousMode);
+                if (callback) {
+                    callback();
+                }
+            }
+        });
+    }
+
+    protected sendLinearPowerCommand (value: number) {
```

Comment:
👍 I like this - but not sure on the naming as this is `DirectWrite` mode and isn't just for setting power - ie. it's used for setting LED color/RGB on a hub.